distribution/std-lib/Standard/src/Image/Codecs.enso

```diff
@@ -0,0 +1,93 @@
+from Standard.Base import all
+import Standard.Base.System.File
+import Standard.Image.Data.Image
+import Standard.Image.Codecs.Internal
+
+polyglot java import org.enso.image.Codecs as Java_Codecs
+
+type Read_Flag
+
+    ## Read image with alpha channel, otherwise it gets cropped.
+    type Read_Alpha_Channel
+
+    ## Always convert image to the single channel grayscale image.
+    type Read_Grayscale
+
+    ## Use the gdal driver for loading the image.
+    type Read_Gdal
+
+type Write_Flag
+
+    ## For JPEG, it can be a quality from 0 to 100 (the higher is the better).
+       Default value is 95.
+    type Write_Jpeg_Quality value
+
+    ## Enable JPEG features. Disabled by default.
+    type Write_Jpeg_Progressive
+
+    ## Enable JPEG features. Disabled by default.
+    type Write_Jpeg_Optimize
+
+    ## Separate luma quality level, 0 - 100, default is 0 - don't use.
+    type Write_Jpeg_Luma_Quality value
+
+    ## Separate chroma quality level, 0 - 100, default is 0 - don't use.
+    type Write_Jpeg_Chroma_Quality value
+
+    ## For PNG, it can be the compression level from 0 to 9. A higher value
+       means a smaller size and longer compression time. Default value is 3.
+    type Write_Png_Compression value
+
+    ## For WEBP, it can be a quality from 1 to 100 (the higher is the better).
+       By default (without any parameter) and for quality above 100 the lossless
+       compression is used.
+    type Write_Webp_Quality value
+
+## Read the image from a file.
+
+   The functon reads images in BGR format, or BGRA if the `Read_Alpha_Channel` flag is specified.
+
+   > Example
+     Read the image.
+         Codecs.read "image.png"
+     Read the image with alpha channel.
+         Codecs.read "image.png" Codecs.Read_Alpha_Channel
+     Read the image and convert it to grayscale.
+         Codecs.read "image.png" Codecs.Read_Grayscale
+read : (Text | File) -> (Read_Flag | Vector) -> Image ! File.Io_Error
+read location flags=[] =
+    path = case location of
+        File.File _ -> location.path
+        _ -> location
+    read_flags = case flags of
+        Vector.Vector _ ->
+            if flags.is_empty then Java_Codecs.READ_FLAG_EMPTY else
+                flags.map Internal.read_flag_to_integer . reduce (_.bit_or _)
+        _ -> Internal.read_flag_to_integer flags
+    Panic.recover (Image.Image (Java_Codecs.read path read_flags)) . catch e->
+        case e of
+            Polyglot_Error _ -> Error.throw (File.Io_Error ('Failed to read ' + path))
```

Comment:
```suggestion
            Polyglot_Error _ -> Error.throw (File.Io_Error ('Failed to read the file at ' + path.to_text + "."))
```
This indicates that you haven't tested the error cases. Please make sure that you have comprehensive test coverage.